internal/zero/reconciler/records.go

```diff
@@ -0,0 +1,132 @@
+package reconciler
+
+// RecordSetBundle is an index of databroker records by type
+type RecordSetBundle[T Record[T]] map[string]RecordSet[T]
```

Comment:
nit: I'm not sure why this is a generic when it's only ever used for `DatabrokerRecord`?

Reply:
I found it convenient to use generics for a higher level functionality like this one, as it allows writing more concise and easy to read tests compared to using the full data structure as it might be pretty large. See `record_tests.go`